packages/flutter/test/material/slider_test.dart

```diff
@@ -908,6 +909,105 @@ void main() {
   });
 
+  testWidgets('Slider onChangeStart and onChangeEnd fire once', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/28115
+
+    int startFired = 0;
+    int endFired = 0;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: GestureDetector(
+                onHorizontalDragUpdate: (_) { },
+                child: Slider(
+                  value: 0.0,
+                  onChanged: (double newValue) { },
+                  onChangeStart: (double value) {
+                    startFired += 1;
+                  },
+                  onChangeEnd: (double value) {
+                    endFired += 1;
+                  },
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.timedDrag(
+      find.byType(Slider),
+      const Offset(20.0, 0.0),
+      const Duration(milliseconds: 100),
+    );
+
+    expect(startFired, equals(1));
+    expect(endFired, equals(1));
+  });
+
+  testWidgets('Slider has proper slops', (WidgetTester tester) async {
+    double value = 0.5;
+    late VoidCallback updateSlider;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: GestureDetector(
+                onHorizontalDragUpdate: (_) { },
+                child: StatefulBuilder(
+                  builder: (BuildContext context, StateSetter setState) {
+                    updateSlider = () => setState(() { });
+                    return Slider(
+                      value: value,
+                      onChanged: (double newValue) {
+                        value = newValue;
+                      },
+                    );
+                  }
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Test when slider drag is active
+
+    final Offset center = tester.getCenter(find.byType(Slider));
+    final TestGesture touchGesture = await tester.startGesture(center);
+
+    /// Wait before start fires
+    await tester.pump(const Duration(milliseconds: 100));
+
+    await touchGesture.moveBy(const Offset(Slider.activeDragSlop + 1.0, 0.0));
+    expect(value, equals(0.5));
+
+    await touchGesture.moveBy(const Offset(1.0, 0.0));
+    expect(value, greaterThan(0.5));
+
+    await touchGesture.up();
+    value = 0.0;
+    updateSlider();
+
+    // Test when slider drag is not yet active
+
+    final TestGesture touchGestureNotActive = await tester.startGesture(center);
+
+    await touchGestureNotActive.moveBy(const Offset(kTouchSlop, 0.0));
+    expect(value, equals(0.0));
+
+    await touchGestureNotActive.moveBy(const Offset(1.0, 0.0));
+    expect(value, greaterThan(0.5));
+  });
```

Comment:
I noticed that other tests do `await gesture.up();` and sometimes `await tester.pumpAndSettle();` at their end. Is this required and why?